elements/pfe-switch/src/pfe-switch.js

```diff
@@ -0,0 +1,178 @@
+import PFElement from "../../pfelement/dist/pfelement.js";
+
+class PfeSwitch extends PFElement {
+  static get tag() {
+    return "pfe-switch";
+  }
+
+  get schemaUrl() {
+    return "pfe-switch.json";
+  }
+
+  get templateUrl() {
+    return "pfe-switch.html";
+  }
+
+  get styleUrl() {
+    return "pfe-switch.scss";
+  }
+
+  static get events() {
+    return {
+      change: `${this.tag}:change`
+    };
+  }
+
+  get pfeChecked() {
+    return this.getAttribute("pfe-checked");
+  }
+
+  // Declare the type of this component
+  static get PfeType() {
+    return PFElement.PfeTypes.Content;
+  }
+
+  constructor() {
+    super(PfeSwitch, { type: PfeSwitch.PfeType });
+    this._updateSwitchState = this._updateSwitchState.bind(this);
+    this.settings = {};
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    const $pfeSwitch = this.shadowRoot.querySelector(".pfe-switch");
+
+    // Get settings from attributes
+    const settingAttributes = [
+      "pfe-message-on",
+      "pfe-message-off",
+      "pfe-hide-label"
+    ];
+    for (let index = 0; index < settingAttributes.length; index++) {
+      const settingAttribute = settingAttributes[index];
+      // Adds settings to an object and lops of `data-` from the index
+      this.settings[settingAttribute.substr(4)] = this.getAttribute(
+        settingAttribute
+      );
+    }
+
+    // Get a reference to the checkbox and label that should be slotted
+    this.checkbox = null;
+    this.label = null;
+    const $assignedElements = this.shadowRoot
+      .querySelector("slot")
+      .assignedElements();
+    for (let index = 0; index < $assignedElements.length; index++) {
+      const $assignedElement = $assignedElements[index];
+      switch ($assignedElement.tagName) {
+        case "LABEL":
+          this.label = $assignedElement;
+          break;
+        case "INPUT":
+          this.checkbox = $assignedElement;
+          break;
+      }
+    }
+
+    if (this.label) {
+      // Get the fallback text and store it as a var
+      this.settings["fallback-text"] = this.label.textContent;
+
+      // Hide the label if necessary
+      if (this.settings["hide-label"]) {
+        this.label.classList.add("pfe-sr-only");
+        const switchIcon = document.createElement("span");
+        switchIcon.classList.add("pfe-switch__checkmark");
+        $pfeSwitch.appendChild(switchIcon);
+      }
+
+      // Check for label text
+      if (!this.label.textContent.trim().length) {
+        console.error(
+          "pfe-switch: Label text not found. Form elements require meaningful label text for accessibility. The label can be visually hidden, but must exist."
```

Comment:
```suggestion
          `${this.tag}: Label text not found. Form elements require meaningful label text for accessibility. The label can be visually hidden, but must exist.`
```

Using the variable helps in case the component is renamed.